.azure-pipelines.yml

```diff
@@ -76,26 +68,17 @@ jobs:
       #       "msvc-v16.10.0" | build
       #     cacheHitVar: BUILD_CACHE_RESTORED
 
-      - task: UsePythonVersion@0
-        inputs:
-          versionSpec: '3.x' 
-          addToPath: true 
-          architecture: "$(TRIPLET)"
-
       - script: |
           git clone https://github.com/microsoft/vcpkg %VCPKG_DIR%
-          git -C %VCPKG_DIR% checkout %VCPKG_REF%
```

Comment:
we'll use master going forward

Reply:
but if we do that then every time master is updated it could potentially update one of our depenencies, which would be bad for several reasons:

1. it blows away our vcpkg cache or at least it should right? the commit hash should be part of the circle cache key since new packaging code could mean new packages
2. blowing away the cache causes us to fetch the deps and build them from source which means very slow builds every time vcpkg/master is updated
3. what happens when a new version of a dependency is no longer compatible with our code. mainlining the latest stuff all the time can totally screw us into a bunch of work to bring our code up to date with whatever library, not to mention could pull in new bugs from those or slowdowns or whatever. its certainly better to pin the versions of our dependencies

i dont know much about vcpkg but these are my concerns, maybe im totally misunderstanding. its better for us to take updates when we want to rather than randomly via external factors. thats already a sucky part of CI we should seek to minimize that. every now and then we can update and validate that the update is good (like we do for thirdparty submodules)

Reply:
convinced:) my reasoning was that _sometimes_ details in the packaging changes (i.e. how they're built, e.g. URLs to sources or so). but that really only happened a few times in the past years.